### PR TITLE
Do not use `javax.mail` for `doCheckAdminAddress`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/6165#discussion_r779674859 for context.

https://stackoverflow.com/q/201323/12916 has plenty of advice but the best I saw was https://stackoverflow.com/a/202528/12916. This is just form validation; you can ignore it if you know better.

Note that this implementation is actually stricter than the status quo in some cases!

```
jshell> new InternetAddress("xxx")
$5 ==> xxx
```

As to requiring a `.`, well my old friend Adam may have been pulling my leg but he claims to have been the sysadmin for Anguilla back in the early days of the Internet and to have received mail as `a@ai`.

### Proposed changelog entries

* N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
Allows stuff like `Foo Bar <foo@bar` which the older implementation would have rejected with e.g. "Missing `>`", but I suppose we can live with that.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
